cms/core/tests/test_management_command_locked_migrate.py

```diff
@@ -0,0 +1,109 @@
+
+from django.core.management import call_command
+from django.test import TestCase
+
+
+class LockedMigrateCommandIntegrationTest(TestCase):
+    def setUp(self):
+        self.buffer = StringIO()
+        patcher = patch("cms.core.management.commands.locked_migrate.call_command", autospec=True)
+        self.mock_call_command = patcher.start()
+        self.addCleanup(patcher.stop)
+
+    def assert_lock_acquired(self, output_buffer=None):
+        buffer = output_buffer or self.buffer
+        output = buffer.getvalue().strip()
+        self.assertIn("Acquiring lock", output)
+        self.assertIn("Lock acquired - running migrations", output)
+
+    def call_locked(self, timeout, skip_unapplied_check=True, stdout_buffer=None):
+        args = ["--timeout", str(timeout)]
+        if skip_unapplied_check:
+            args.append("--skip-unapplied-check")
+
+        call_command("locked_migrate", *args, stdout=stdout_buffer or self.buffer)
+
+    def _setup_lock_side_effect(self):
+        """Configure the mock to signal when the lock is held and simulate migration time."""
+        lock_held = threading.Event()
+
+        def side_effect(*args, **kwargs):
+            lock_held.set()
+            time.sleep(1.5)
+
+        self.mock_call_command.side_effect = side_effect
+        return lock_held
+
+    def test_skip_unapplied_check_acquires_lock_and_runs_migrations(self):
+        """When skipping unapplied check, it should always acquire the lock and run migrations."""
+        self.call_locked(timeout=1, skip_unapplied_check=True)
+        self.assert_lock_acquired()
+
+    @patch("cms.core.management.commands.locked_migrate.Command._has_unapplied_migrations")
+    def test_has_unapplied_migration(self, mock_has_unapplied):
+        """When there are unapplied migrations, it should acquire the lock and run migrations."""
+        for has_pending in (True, False):
+            with self.subTest(has_pending=has_pending):
+                mock_has_unapplied.return_value = has_pending
+                self.mock_call_command.reset_mock()
+                self.buffer = StringIO()
+
+                self.call_locked(timeout=1, skip_unapplied_check=False)
+                if has_pending:
+                    self.assert_lock_acquired()
+                else:
+                    self.mock_call_command.assert_not_called()
+
+    def test_lock_timeout_under_concurrency(self):
+        """Second thread should exit(1) if it can't get the lock in time."""
+        thread_1_output = StringIO()
+        thread_2_output = StringIO()
+
+        lock_held = self._setup_lock_side_effect()
+
+        # Start thread1 and wait until it grabs the lock
+        thread1 = threading.Thread(target=lambda: self.call_locked(5, stdout_buffer=thread_1_output))
+        thread1.start()
+        lock_held.wait(timeout=1)
+
+        with self.assertRaises(SystemExit) as cm:
+            self.call_locked(timeout=1, stdout_buffer=thread_2_output)
```

Comment:
Nit-pick: In this code we use `call_locked` in an inconsistent manner. Over here we explicitly provide `timeout=1` while in other places we don't.

Reply:
Now consistent [`46b529e` (#324)](https://github.com/ONSdigital/dis-wagtail/pull/324/commits/46b529e8cee12f43519f56a701271ab77c37b6d7)